presto-hive-common/src/main/java/com/facebook/presto/hive/CacheQuota.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import io.airlift.units.DataSize;
+
+import java.util.Optional;
+
+import static com.facebook.presto.hive.CacheQuotaScope.GLOBAL;
+import static com.google.common.hash.Hashing.md5;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class CacheQuota<T>
```

Comment:
This class has to be serializable. I can't quite recall if generic T can be serialized. (Maybe it could; but I can't find a good example). If it's too complicated, just make `identity` of type `String` so we don't have to worry about it.